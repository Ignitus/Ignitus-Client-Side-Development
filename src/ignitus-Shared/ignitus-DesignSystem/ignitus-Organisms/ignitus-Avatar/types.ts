export type avatar = {
  name: string;
  src: string;
};

/** As AvatarImage represents img feel free to add more props here, as <img /> supports lot of other attributes. */
export type AvatarImageProps = {
  src: string;
  alt: string;
};
