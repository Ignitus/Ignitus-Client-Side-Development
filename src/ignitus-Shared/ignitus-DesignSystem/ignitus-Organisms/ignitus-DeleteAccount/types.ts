export interface PropTypes {
  username: string;
}
