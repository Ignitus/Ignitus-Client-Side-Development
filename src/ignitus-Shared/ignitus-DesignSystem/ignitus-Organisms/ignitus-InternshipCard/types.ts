export type CardTypes = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type?: string;
  JobType?: string;
};

export type CardProps = {
  ImageLink: string;
  Title: string;
  Position?: string;
  Place?: string;
  Company?: string;
  Description?: string;
  type: string;
  JobType?: string;
};

export type StyleProps = {
  size?: string;
};
