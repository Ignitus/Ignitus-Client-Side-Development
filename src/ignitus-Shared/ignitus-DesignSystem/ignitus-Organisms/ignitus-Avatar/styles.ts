import styled from '@emotion/styled';

export const AvatarImage = styled.img`
  max-width: 9rem;
`;
