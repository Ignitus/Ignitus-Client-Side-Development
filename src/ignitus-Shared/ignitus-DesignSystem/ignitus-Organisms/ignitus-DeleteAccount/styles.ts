import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { DefaultCheckbox } from '../../..';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  padding: 2rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 40rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  justify-content: flex-end;
  margin-top: 2rem;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 1rem 0;
`;

export const StyledCheckbox = styled(DefaultCheckbox)`
  font-size: ${F.SM};
  margin-bottom: 1rem;
`;

export const StyledTextArea = styled.textarea`
  background-color: ${C.GreyLight};
  padding: 1rem;
  color: ${C.GreySecondaryText};
  font-weight: ${F.SemiBold};
  font-size: ${F.MD};
  border: none;
  border-radius: 1rem;

  ::placeholder {
    color: ${C.GreySecondaryText};
    font-weight: ${F.SemiBold};
    font-size: ${F.MD};
    opacity: 0.5;
  }
`;

export const StyledRoundedButton = styled(RoundedButton)`
  font-weight: ${F.Bold};
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
