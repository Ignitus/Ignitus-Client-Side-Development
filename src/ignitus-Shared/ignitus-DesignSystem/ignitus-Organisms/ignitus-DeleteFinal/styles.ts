import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import { Paragraph } from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';
import { Icon } from '../../../ignitus-Utilities/Components/icon';
import { RoundedButton } from '../../ignitus-Atoms/buttons';

export const DeleteContainer = styled.div`
  background-color: ${C.White};
  background-size: contain;
  border-radius: 1rem;
  display: inline-flex;
  flex-direction: column;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  width: 45rem;
  padding: 2rem;
`;

export const TopSection = styled(flexibleRowDiv)`
  justify-content: space-between;
`;

export const MiddleSection = styled.div`
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
`;

export const BottomSection = styled(flexibleRowDiv)`
  margin-top: 1rem;
  justify-content: flex-end;
`;

export const StyledIcon = styled(Icon)`
  width: 1.4rem;
  fill: ${props => props.color};
  transition: 0.3s;
  flex: 1;
  margin-top: 0.2rem;
`;

export const IconContainer = styled.button`
  border: none;
  cursor: pointer;
  overflow: hidden;
  margin: 0 0.2rem;
  height: 2.8rem;
  width: 2.8rem;
  border-radius: 50%;
  background-color: ${props => props.color};
  outline-color: transparent;
`;

export const StyledParagraph = styled(Paragraph)`
  padding: 0.5rem 0;
`;

export const StyledText = styled(Paragraph)`
  font-weight: ${F.Bold};
  padding: 0.5rem 0;
`;

export const StyledRoundedButton = styled(RoundedButton)`
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
`;
