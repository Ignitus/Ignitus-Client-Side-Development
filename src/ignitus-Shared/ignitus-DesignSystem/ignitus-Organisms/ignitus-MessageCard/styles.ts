import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import * as T from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv } from '../../shared';

export const MessageContainer = styled(flexibleRowDiv)`
  border-radius: 1rem;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  box-sizing: border-box;
  padding: 1rem;
  width: 100%;
  margin-bottom: 1rem;
`;

export const ImageContainer = styled.div`
  position: relative;
  padding: 0.5rem;

  img {
    border-radius: 50%;
    width: 4rem;
    max-width: 100%;
    max-height: 100%;
  }

  svg {
    position: absolute;
    bottom: 0;
    right: 0;
    stroke: ${C.White};
    fill: #219653;
    height: 2rem;
    width: 2rem;
  }
`;

export const MessageSubTitle = styled(T.Paragraph)`
  margin-left: 1rem;
`;

export const Sender = styled(T.Paragraph)`
  font-weight: ${F.Bold};
  font-size: ${F.MD};
  color: ${C.IgnitusBlue};
  margin-bottom: 0.5rem;
`;

export const SenderPosition = styled(T.Paragraph)`
  font-weight: ${F.Normal};
  font-size: ${F.SM};
  color: ${C.IgnitusBlue};
  margin-bottom: 0.5rem;
`;

export const MessageDateTitle = styled.div`
  text-align: right;
  margin-left: auto;
`;

export const Details = styled(T.Paragraph)`
  font-family: Open Sans;
  font-weight: ${F.Normal};
  font-size: ${F.SM};
  color: ${C.GreyOne};
  margin-bottom: 0.5rem;
`;
