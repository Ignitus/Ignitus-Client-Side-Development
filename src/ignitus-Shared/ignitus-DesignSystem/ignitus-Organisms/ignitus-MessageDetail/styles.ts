import styled from '@emotion/styled';
import * as C from '../../ignitus-Atoms/colors';
import * as T from '../../ignitus-Atoms/typography';
import * as F from '../../ignitus-Atoms/fonts';
import { flexibleRowDiv, flexibleColDiv, Paragraph } from '../../shared';
import { DefaultButtonWithIcon } from '../../..';

export const MessageContainer = styled(flexibleColDiv)`
  border-radius: 1rem;
  box-shadow: 0 2px 4px 0 ${C.boxShadowColor};
  background-color: ${C.White};
  box-sizing: border-box;
  width: 100%;
  margin-bottom: 1rem;
`;

export const ImageContainer = styled.div`
  position: relative;
  padding: 0.5rem;

  img {
    border-radius: 50%;
    width: 4rem;
    max-width: 100%;
    max-height: 100%;
  }

  svg {
    position: absolute;
    bottom: 0;
    right: 0;
    stroke: ${C.White};
    fill: #219653;
    height: 2rem;
    width: 2rem;
  }
`;

export const MessageSubTitle = styled(T.Paragraph)`
  margin-left: 1rem;
`;

export const Sender = styled(T.Paragraph)`
  font-weight: ${F.Bold};
  font-size: ${F.MD};
  color: ${C.IgnitusBlue};
  margin-bottom: 0.5rem;
`;

export const SenderPosition = styled(T.Paragraph)`
  font-weight: ${F.Normal};
  font-size: ${F.SM};
  color: ${C.IgnitusBlue};
  margin-bottom: 0.5rem;
`;

export const MessageDateTitle = styled.div`
  text-align: right;
  margin-left: auto;
`;

export const Details = styled(T.Paragraph)`
  font-family: Open Sans;
  font-weight: ${F.Normal};
  font-size: ${F.SM};
  color: ${C.GreyOne};
  margin-bottom: 0.5rem;
`;

export const Message = styled(Paragraph)``;

export const TopSection = styled(flexibleRowDiv)`
  box-sizing: border-box;
  padding: 1rem 1rem 0 1rem;
  width: 100%;
`;

export const MiddleSection = styled(flexibleRowDiv)`
  box-sizing: border-box;
  padding: 1rem;
  width: 100%;
`;

export const BottomSection = styled(flexibleRowDiv)`
  box-sizing: border-box;
  padding-right: 1rem;
  width: 100%;
  margin-bottom: 1rem;
  justify-content: flex-end;
`;

export const StyledDefaultButtonWithIcon = styled(DefaultButtonWithIcon)`
  flex-direction: row-reverse;
  box-shadow: 0 6px 12px ${C.boxShadowColorLight};
  border-radius: 1rem;
  font-weight: ${F.Bold};
  svg {
    margin-right: 0.2rem;
  }
`;

export const StyledLine = styled.hr`
  width: -webkit-fill-available;
`;
