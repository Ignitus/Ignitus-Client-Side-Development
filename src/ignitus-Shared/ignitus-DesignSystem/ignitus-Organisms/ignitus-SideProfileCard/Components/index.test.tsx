import React from 'react';
import { mount, render } from 'enzyme';
import emotionSerializer, { matchers } from 'jest-emotion';
import { SideProfileCard } from './index';
import { Props } from '../types';
import * as S from '../styles';
import { Button } from '../../../ignitus-Atoms/buttons';

const props: Props = {
  name: 'Helen',
  designation: 'Student',
  trackingInformation: '10 students are tracking your progree',
};

expect.addSnapshotSerializer(emotionSerializer);
expect.extend(matchers);

it('should rendered', () => {
  const tree = render(<SideProfileCard {...props} />);
  expect(tree).toMatchSnapshot();
});

describe('<SideProfileCard />', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = mount(<SideProfileCard {...props} />);
  });

  it('should render name, designation, trackingInformation correctly', () => {
    expect(wrapper.find(S.Name).text()).toEqual(props.name);
    expect(wrapper.find(S.Designation).text()).toEqual(props.designation);
    expect(wrapper.find(S.Text).text()).toEqual(props.trackingInformation);
  });

  it('should render button', () => {
    expect(wrapper.find(Button).length).toEqual(1);
  });

  it('should render avatar', () => {
    expect(wrapper.find(S.Avatar).length).toEqual(1);
  });
});
