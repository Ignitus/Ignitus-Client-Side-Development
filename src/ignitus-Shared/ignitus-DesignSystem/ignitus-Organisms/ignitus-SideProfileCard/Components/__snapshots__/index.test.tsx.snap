// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should rendered 1`] = `
.emotion-11 {
  background-color: #ffffff;
  border-radius: 1rem;
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
  color: #000066;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  height: 12rem;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  padding: 1.5rem;
  width: 17rem;
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  margin-bottom: 2rem;
}

.emotion-0 {
  border-radius: 1rem;
  height: 6rem;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  text-align: center;
}

.emotion-3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  line-height: 1.7rem;
}

.emotion-1 {
  font-weight: 500;
  font-size: 22px;
  color: #000066;
}

.emotion-2 {
  font-weight: 500;
  font-size: 16px;
  color: #000066;
}

.emotion-4 {
  border-radius: 5px;
  border: none;
  cursor: pointer;
  outline: none;
  padding: 16px 32px;
  text-align: center;
  line-height: 1rem;
  margin: 0.5rem;
  outline: none;
  font-size: 18px;
  font-weight: 400;
  -webkit-transition: all 0.3s ease;
  transition: all 0.3s ease;
  outline: none;
  color: #ffffff;
  background-color: #000066;
  padding: 8px 20px;
  font-size: 16px;
  border-radius: 3px;
  border: none;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-4:hover,
.emotion-4:focus {
  box-shadow: 0 2px 4px 0 rgba(0,0,0,0.05);
}

.emotion-4 a {
  color: #ffffff;
}

.emotion-4:hover,
.emotion-4:focus {
  background-color: #F9BF64;
  color: #000066;
  outline: none;
}

.emotion-4:hover a,
.emotion-4:focus a {
  color: #000066;
}

.emotion-4:hover svg,
.emotion-4:focus svg {
  fill: #000066;
}

.emotion-10 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.emotion-8 {
  fill: #000066;
  height: 2rem;
  margin: 0 1rem 0 0;
}

.emotion-9 {
  font-size: 16px;
  font-weight: 400;
  color: #2D2D2D;
  margin: 0;
}

<div
  class="emotion-11"
>
  <div
    class="emotion-7"
  >
    <img
      alt="ProfileImage"
      class="emotion-0"
      src="https://storage.googleapis.com/ignitus_assets/ig-avatars/melanie.png"
    />
    <div
      class="emotion-6"
    >
      <div
        class="emotion-3"
      >
        <h3
          class="emotion-1"
        >
          Helen
        </h3>
        <h6
          class="emotion-2"
        >
          Student
        </h6>
      </div>
      <div
        class="emotion-5"
      >
        <button
          class="emotion-4"
        >
           ViewProfile 
        </button>
      </div>
    </div>
  </div>
  <div
    class="emotion-10"
  >
    <svg
      aria-label="Bookmark Icon"
      class="emotion-8"
      viewBox="0 0 384 512"
      xmlns="http://www.w3.org/2000/svg"
    >
      <title>
        Bookmark Icon
      </title>
      <path
        d="M336 0H48C21.49 0 0 21.49 0 48v464l192-112 192 112V48c0-26.51-21.49-48-48-48zm0 428.43l-144-84-144 84V54a6 6 0 0 1 6-6h276c3.314 0 6 2.683 6 5.996V428.43z"
      />
    </svg>
    <p
      class="emotion-9"
    >
      10 students are tracking your progree
    </p>
  </div>
</div>
`;
